### ISSUE TITLE
[BUG] Annotated attributes of cclass are not supporting pointers

### ISSUE THREAD

**Issue by matusvalo** [cython/cython#4293]
**Describe the bug**
The compilation is failing, when attribute of cclass is declared using annotated type containing pointer.

**To Reproduce**
Following code:
```python
import cython

@cython.cclass
class Foo:
    a: cython.pointer(cython.int)

    def bar(self):
        self.a = cython.NULL
```
fails during compilation with error:
```
$ cython -3 test.py

Error compiling Cython file:
------------------------------------------------------------
...
@cython.cclass
class Foo:
    a: cython.pointer(cython.int)

    def bar(self):
        self.a = cython.NULL
                      ^
------------------------------------------------------------

test.py:8:23: Cannot convert 'void *' to Python object
```

**Expected behavior**
Compilation should be successfull.

**Environment (please complete the following information):**
 - OS: Linux
 - Python version: Python 3.9.2
 - Cython version: master

**Additional context**
When `declare()` statement or `cython.p_int` type is used, compilation is successful:

```python
import cython

@cython.cclass
class Foo:
    a = cython.declare(cython.pointer(cython.int))

    def bar(self):
        self.a = cython.NULL
```

```python
import cython

@cython.cclass
class Foo:
    a: cython.p_int

    def bar(self):
        self.a = cython.NULL
```

**Comment by scoder** [comment]
This is a know issue and something that should work with the dataclasses PR #3400.
https://github.com/cython/cython/blob/9f78ed075f7371f931c7fa76b45589841c25668f/Cython/Compiler/ExprNodes.py#L2037-L2038

**Comment by da-woods** [comment]
I don't think this is fixed by #3400? I get the same issue on the dataclass branch (and the same issue again if I mark it as a Cython dataclass, just to check that this wasn't making a difference).

If I remove the function with the NULL assignment then it compiles to

```
struct __pyx_obj_9ptr_annot_Foo {
  PyObject_HEAD
  PyObject *a;
};
```

(i.e. not an `int*` pointer).

So I don't think there's anything I can cherry-pick for a quick solution to this...

**Comment by da-woods** [comment]
The warning:

```
 warning(annotation.pos, "Annotation ignored since class-level attributes must be Python objects. " 
         "Were you trying to set up an instance attribute?", 2) 
```

Is catching things like:

```
cdef class X:
    x : cython.int = 5
```

which are treated differently in regular classes and dataclasses. But it's the assignment at the point of definition that's relevant here.

**Comment by matusvalo** [comment]
This bug can be closed since it seems working on master branch.
```
(cython39) matus@MacBook-Pro:~/dev/cython/test$ cat test.py
import cython

@cython.cclass
class Foo:
    a: cython.pointer(cython.int)

    def bar(self):
        self.a = cython.NULL
(cython39) matus@MacBook-Pro:~/dev/cython/test$ cython -3 test.py
(cython39) matus@MacBook-Pro:~/dev/cython/test$
```

**Comment by da-woods** [comment]
For information, this was fixed in cb5eb5a139cbefddffa54a7bddf64b3a03f349fd

**Comment by matusvalo** [comment]
>(so actually you reported this bug twice... nevermind!)

¯\_(ツ)_/¯
